### DISCUSSION TITLE
The correct approach to restart closed/broken connections?

### DISCUSSION THREAD

**Discussion by ZzZombo** [frankie567/httpx-ws#28] [Q&A]
Hello again. I'm currently stuck here: if the remote end suddenly terminates their end of the connection, how do I cleanly restart it?

My current flow is as follows, for context:

* First, an infinite `asyncio` loop is established that puts all incoming WS messages into a dictionary.
* When an external event is handled, a different coroutine sends a corresponding WS message, then awaits in a loop until the dictionary contains the response to the sent message, then handles the response in some way and returns the result to the caller.
* The calling coroutine waits for results, it's undesirable for it to proceed without. Currently it's only allowed if an exception is thrown, and I'd like to keep it this way.

The remote end is a server we control, but currently it's going through a rewrite so understandably it tends to crash while all kinks are being ironed out. The WS server gets spin up within moments for the most part after termination, so ideally we should wait until it's ready, reinstate the connection and try again. But the most important part is that the calling coroutine can't be asked to retry again later, it must get the result on its first and only invocation, although it can wait a reasonable amount of time for it.

Any thoughts? Thanks in advance.

**Comment by frankie567** [comment]
I'm not sure I follow you entirely, but I guess you should be able to wrap your main loop wyth a `try..except` to catch [`WebSocketDisconnect`](https://frankie567.github.io/httpx-ws/reference/httpx_ws/#httpx_ws.WebSocketDisconnect) and [`WebSocketNetworkError`](https://frankie567.github.io/httpx-ws/reference/httpx_ws/#httpx_ws.WebSocketNetworkError) exceptions. When this happen, you can restart the connection and all your logic.
